extensions/notion/src/utils/openPage.ts

```diff
@@ -2,30 +2,25 @@ import { closeMainWindow, getPreferenceValues, getApplications, open } from "@ra
 
 import { Page } from "./notion/page";
 
-export let openIn = "web";
-export let isNotionInstalled = false;
+let openIn = getPreferenceValues<Preferences>().open_in;
 
-export async function checkOpenInApp() {
-  const preferences = getPreferenceValues<Preferences>();
-
-  const installedApplications = await getApplications();
-  isNotionInstalled = installedApplications.some((app) => app.bundleId === "notion.id");
-
-  if (preferences.open_in === "app") {
-    openIn = isNotionInstalled ? "app" : "web";
-  } else {
-    openIn = "web";
-  }
+let hasCheckedDefault = false;
+export async function checkedDefaultOpenMethod() {
+  if (hasCheckedDefault) return;
+  hasCheckedDefault = true;
+  const apps = await getApplications();
+  const defaultApp = apps.find((app) => app.name === "Notion");
+  if (defaultApp) openIn = defaultApp;
```

Comment:
I believe this is the culprit for the bug I was talking about.

Reply:
Bummer, really curious why this is not happening for me.

Reply:
@thomaslombart I wonder if it is related to this notion setting. Going to do some testing

![CleanShot 2023-12-21 at 23 10 38@2x](https://github.com/raycast/extensions/assets/65966422/5510b94c-81b7-4112-8969-5aae78e1e041)

Reply:
Yep, you are right, and I finally was able to reproduce your behavior. pushing a fix. thanks